core/src/lib/tree.h

```diff
@@ -52,42 +52,42 @@ struct s_mem {
         ((TREE_NODE *)(node->child.first())
 
 struct delta_list {
-  struct delta_list* next;
+  delta_list* next;
   JobId_t JobId;
   int32_t FileIndex;
 };
 
+/* type values */
+enum class TreeNodeType
+{
+  UNKNOWN = 0,
+  ROOT,     // root node
+  NEWDIR,   // created directory to fill path
+  DIR,      // directory entry
+  DIR_NLS,  // directory -- no leading slash -- win32
+  FILE,     // file entry
+};
+
 /**
  * Keep this node as small as possible because
  *   there is one for each file.
  */
 struct s_tree_node {
-  s_tree_node()
-      : type{false}
-      , extract{false}
-      , extract_dir{false}
-      , hard_link{false}
-      , soft_link{false}
-      , inserted{false}
-      , loaded{false}
-  {
-  }
   /* KEEP sibling as the first member to avoid having to
    *  do initialization of child */
   rblink sibling;
   rblist child;
-  char* fname{};                /* file name */
-  int32_t FileIndex{};          /* file index */
-  uint32_t JobId{};             /* JobId */
-  int32_t delta_seq{};          /* current delta sequence */
-  uint16_t fname_len{};         /* filename length */
-  unsigned int type : 8;        /* node type */
-  unsigned int extract : 1;     /* extract item */
-  unsigned int extract_dir : 1; /* extract dir entry only */
-  unsigned int hard_link : 1;   /* set if have hard link */
-  unsigned int soft_link : 1;   /* set if is soft link */
-  unsigned int inserted : 1;    /* set when node newly inserted */
-  unsigned int loaded : 1;      /* set when the dir is in the tree */
+  std::string fname{};                      /* file name */
```

Comment:
note: this also creates a memory leak, as fname is usually `tree_alloc`ed, so changing it to `std::string` messes things up